### PR TITLE
feat: enable Triton kernels on MUSA

### PR DESCRIPTION
**Pull request by Arlo-mt**
## Summary

Thank you for merging the previous PR that added the MUSA backend support. Building on that foundation, this PR adds MUSA support for Triton-backed RL-Kernel operators.

MUSA tensors are not reported as CUDA tensors by PyTorch, so the existing
`.is_cuda` and `torch.cuda.is_available()` checks could incorrectly reject
MUSA execution or select the CPU fallback. This change adds explicit MUSA
device handling while preserving the existing CUDA and ROCm paths.

## Changes

- Add MUSA device detection through `torch_musa` and `torch.musa`.
- Add the MUSA device and backend constants.
- Update Triton operator validation to accept MUSA tensors.
- Enable MUSA Triton dispatch for:
  - `logp`
  - `grpo_loss`
  - `ratio_kl`
  - `linear_logp`
  - `det_gemm`
  - `batch_invariant_logp`
  - `rms_norm`
  - `embedding`
  - `silu`
  - `swiglu`
  - `rope`
- Keep PyTorch fallback paths available when a Triton backend is unavailable.
- Add MUSA-specific dispatch and `linear_logp` forward/backward coverage.
- Update accuracy and dispatch tests to distinguish MUSA from CPU and CUDA.

This PR does not add MUSA native C++/MUSA kernels. Native MUSA kernel support
is handled separately in the `MUSA-support-native-kernels` branch.

## Validation

Validated locally on:

- GPU: Moore Threads S5000
- PyTorch: `2.9.0.post1+musa5.1.2mp22`
- Triton: `3.2.0`
- MUSA architecture: `mp_22`

Results:

- MUSA Triton operator smoke tests: `5/5 passed`
- MUSA dispatch and accuracy tests: `7/7 passed`
- CPU-safe and dispatch unit tests: `100 passed`
- Python compilation and whitespace checks passed

## Compatibility

- CUDA and ROCm-specific dispatch paths are unchanged.
- CUDA-only tests such as SM90, NCCL, and CUDA extension tests remain
  CUDA-specific.

## Summary by CodeRabbit

- **New Features**
  - Added MUSA support across embedding, loss functions, normalization, rotary embeddings, and related GPU operations.
  - Added deterministic linear log-probability calculations, including tensor-parallel support.
  - MUSA dispatch now selects validated optimized backends with PyTorch fallbacks where needed.
  - Added support for contract-aware attention selection on MUSA.

- **Bug Fixes**
  - Improved numerical consistency for SiLU/SwiGLU and log-probability calculations across supported accelerators.
  - Added validation to prevent matrix multiplication inputs from residing on different devices.

### CODE REVIEW

File: rl_engine/kernels/ops/triton/loss/grpo_loss.py [hunk 1]
```diff
@@ -145,8 +145,10 @@ def group_advantages(
         eps: float = 1e-6,
     ) -> torch.Tensor:
         """Per-sequence reward normalization, computed by the Triton group kernel."""
-        if not rewards.is_cuda:
-            raise RuntimeError("TritonGRPOLossOp requires CUDA tensors.")
+        if rewards.device.type not in ("cuda", "hip", "xpu", "musa"):
```
Comment (zhangj1an) [inline rl_engine/kernels/ops/triton/loss/grpo_loss.py]: For all edited files on this branch, please only add for musa, remove annotation for hip, xpu. We have not officially supported these 2 hardwares haha

**Review by zhangj1an — APPROVED** [review]
Thanks! The change is well-scoped (musa priority map only, fallbacks preserved) and I've verified it doesn't affect existing CUDA/NPU/CPU dispatch.

**Review by Flink-ddd — APPROVED** [review]
LGTM, Thank you.
